### ISSUE TITLE
crashes with widget "anotherone"

### ISSUE THREAD

**Issue by gimplyworxs** [atareao/my-weather-indicator#98]
Since yesterday's update, the app crashes when I try to use the widget design "anotherone".
(Some of the others don't work either btw).
Got a Traceback:

```
Traceback (most recent call last):
  File "/opt/extras.ubuntu.com/my-weather-indicator/bin/my-weather-indicator", line 36, in <module>
    main()
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/myweatherindicator.py", line 917, in main
    MWI()
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/myweatherindicator.py", line 139, in __init__
    self.load_preferences()
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/myweatherindicator.py", line 339, in load_preferences
    self.update_weather()
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/myweatherindicator.py", line 154, in update_weather
    self.update_menu(i)
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/myweatherindicator.py", line 672, in update_menu
    self.widgets[index].set_weather(weather)
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/weatherwidget.py", line 157, in set_weather
    self.parse_data()
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/weatherwidget.py", line 490, in parse_data
    text = text.replace('$PRESSURE$', self.weather_data['current_conditions']['pressure'])
TypeError: replace() argument 2 must be str, not float
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 153, in apport_excepthook
    with os.fdopen(os.open(pr_filename,
FileNotFoundError: [Errno 2] Datei oder Verzeichnis nicht gefunden: '/var/crash/_opt_extras.ubuntu.com_my-weather-indicator_bin_my-weather-indicator.1000.crash'

Original exception was:
Traceback (most recent call last):
  File "/opt/extras.ubuntu.com/my-weather-indicator/bin/my-weather-indicator", line 36, in <module>
    main()
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/myweatherindicator.py", line 917, in main
    MWI()
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/myweatherindicator.py", line 139, in __init__
    self.load_preferences()
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/myweatherindicator.py", line 339, in load_preferences
    self.update_weather()
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/myweatherindicator.py", line 154, in update_weather
    self.update_menu(i)
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/myweatherindicator.py", line 672, in update_menu
    self.widgets[index].set_weather(weather)
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/weatherwidget.py", line 157, in set_weather
    self.parse_data()
  File "/opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/weatherwidget.py", line 490, in parse_data
    text = text.replace('$PRESSURE$', self.weather_data['current_conditions']['pressure'])
TypeError: replace() argument 2 must be str, not float
```

**Comment by gimplyworxs** [comment]
Resolved problem by changing line 490
in the file /opt/extras.ubuntu.com/my-weather-indicator/share/my-weather-indicator/weatherwidget.py
to 
text = text.replace('$PRESSURE$', str(self.weather_data['current_conditions']['pressure']))

**Comment by atareao** [comment]
Fixed in my-weather-indicator - 0.10.1-0extras23.04.00

**Comment by gimplyworxs** [comment]
Thanks a lot for fixing.
